reroll-initiative/old/reroll-initiative.js

```diff
@@ -0,0 +1,260 @@
+                    //console.log("Reroll-Initiative: Round incremented - rerolling initiative")
+                    this._resetAndReroll(combat);
+                }
+            }
+            else {
+                console.log("Rerolling Initiative is currently turned off")
+            }
+            
+        }); 
+    }
+
+    /**
+     * For the given combat instance, call the resetAll method and the rollAll method
+     * @param {Object} combat A Combat instance
+     */
+    async _resetAndReroll(combat){
+        await combat.resetAll();
+        combat.rollAll();
+    }
+}
+
+/**
+ * @class RerollInitiativeConfig
+ * @description Handles the configuration form for the module. Currently inserts within Combat Tracker Config
+ * @todo: tear down all the setting updates here and use the RerollInitiative class instead
+ */
+class RerollInitiativeConfig {
+
+    constructor(){
+        this._hookRenderCombatTrackerConfig();
+    }    
+
+    /**
+     * Hooks on the render of combat tracker config and insert the module config
+     */
+    _hookRenderCombatTrackerConfig(){
+        Hooks.on("renderCombatTrackerConfig", (app, html) => {
+            
+            let settings = RerollInitiative.querySettings();
+            console.log(settings);
+            const targetSetting = "reroll";
+
+            if(html){
+                let rriCheckbox = this._injectCheckboxFormgroup(html);
+
+                if(rriCheckbox){
+                    //Set the state of the checkbox to match the current module reroll setting
+                    rriCheckbox.prop("checked",settings.reroll);
+
+                    // Adjust the window height to allow for new content
+                    app.setPosition({height: app.position.height + 60});
+                
+                    // Handle form submission
+                    //todo: break this out into a separate method?
+                    const submit = html.find('button[type="submit"]');
+                    const form = submit.parent();
+                    form.on("submit", ev => {
+                        
+                        //Retrieve the checkbox state
+                        let rriCheckboxValue = rriCheckbox.prop("checked");
+                        console.log("submit", ev);
+                        console.log("rriCheckbox is: ",rriCheckbox.prop("checked"));
+
+                        //Update target setting with new value
+                        //todo: this needs to be async. break out into new method?
+                        //this.rri.updateOption("reroll", rriCheckboxValue);
+                        RerollInitiative.updateSetting(targetSetting,rriCheckboxValue);
+                    });
+                }
+            }
+        });
+    }
+
+    /**
+     * Injects a checkbox formgroup inside the designated element
+     * @param {*} html
+     * @returns {Object} checkbox The checkbox element that was injected
+     */
```

Comment:
This function is doing to much. Id separate the logic to inject and the logic to return.